agent/consul/autopilot.go

```diff
@@ -52,6 +52,21 @@ func (d *AutopilotDelegate) NotifyState(state *autopilot.State) {
 		} else {
 			metrics.SetGauge([]string{"autopilot", "healthy"}, 0)
 		}
+	} else {
+		/* todo -- re-evaluate metrics emitting NaN
+		From go-metrics v0.3.6, we have stopped resetting to NaN metrics that have
+		no observation in a collection window. A list of such metrics can be found here:
+		https://github.com/hashicorp/consul/issues/10730#issuecomment-929844513
+		To "fix" the metrics and have all appear as NaN and not 0 in prometheus
+		on d.NotifyState() here we could take a pass over all of these metrics and NaN them out.
+		However, the only documented NaN metric is consul_autopilot_healthy so we are
+		only "fixing" this for now.
+		*/
```

Comment:
This is helpful context for the PR review, but might not age well in the code. Might be better as a PR comment.